extras/gedit/conky.lang

```diff
@@ -0,0 +1,716 @@
+         for config files or alternatively conky_your_config_name
+         Unfortunately this will also trigger the syntax highlighting for this document, 
+         you can always manually set this one back to XML.
+         If you have a favourite custom format you can change the globs property to your
+         personal preference. -->
+	<property name="globs">*conky*</property>
+	<property name="block-comment-start">--[[</property>
+	<property name="block-comment-end">]]</property>
+  </metadata>
+
+  <styles>
+	<style id="keyword"	_name="Config keywords"		map-to="def:keyword"/>
+	<style id="attribute"	_name="Text keywords"		map-to="def:type"/>
+	<style id="comment"	_name="Comment"			map-to="def:comment"/>
+	<style id="path" 	_name="Paths and URLs"		map-to="def:string"/>
+	<style id="boolean"	_name="Boolean"			map-to="def:boolean"/>
+	<style id="argument"	_name="Arguments"		map-to="def:preprocessor"/>
+	<style id="brackets"	_name="Brackets ${ ... }"	map-to="def:identifier"/>
+	<style id="decimal"	_name="Decimal"			map-to="def:decimal"/>
+  </styles>
+
+  <definitions>
+   <context id="conkyrc" class="no-spell-check">
+      <include>
+        <context ref="block-comment"/>
+        <context ref="line-comment"/>
+ 	<context ref="path"/>
+       <context ref="brackets"/>
+        <context ref="colors"/>
+        <context ref="fonts"/>
+        <context ref="template"/>
+        <context ref="date"/>
+	<context ref="number"/>
+        <context ref="keywordsConfig"/>
+	<context ref="keywordsText"/>
+	<context ref="boolean"/>
+        <context ref="predefinedColors"/>
+	<context ref="windowType"/>
+	<context ref="windowHints"/>
+	<context ref="alignment"/>
+	<context ref="argumentsText"/>
+	<context ref="argumentsConfig"/>
+      </include>
+    </context>
+
+<!-- ======================================================================================= -->
+<!-- Keep in mind gedit does not like < and >, use &lt; and &gt; instead! -->
+
+    <!-- Block comment, lua syntax -->
+    <context id="block-comment" style-ref="comment" class="comment" class-disabled="no-spell-check">
+      <start>--\[(=*)\[</start>
+      <end>]\%{1@start}]</end>
+      <include>
+	<context ref="def:in-comment"/>
+      </include>
+    </context>
+
+    <!-- Line comment, lua syntax -->
+    <!-- Note: this will only highlight line comments BEFORE the conky.text section -->
+    <context id="line-comment" style-ref="comment" class-disabled="no-spell-check">
+      <match>(?&lt;!conky.text)(\-\-.*)</match>
+    </context>
+
+    <!-- File paths and URLs, starting with https://, http://, ~/ or / -->
+    <context id="path" style-ref="path" class="string">
+      <match>((?:\~\/|\/|htt(?:p|ps)\:\/\/){1}[^\s\}\']+)</match>
+    </context>
+
+    <!-- Brackets --> 
+    <!-- BUG: this will also highlight the last bracket from the conky.config section -->
+    <context id="brackets" style-ref="brackets">
+      <match>(\$\{|(?&lt;!\\)\})</match>
```

Comment:
Alternatively you could use something like `(\$\{|(?<!\\)(\{|\})|\[\[|\]\])` to match *all* brackets (including `[[` and `]]`)